src/main/java/com/petros/bring/bean/factory/AnnotationBeanFactory.java

```diff
@@ -59,14 +73,14 @@ private <T> T createBean(Class<T> beanType) {
             } else {
                 obj = beanType.getConstructor().newInstance();
             }
+            rootContextMap.put(beanDefinition.getName(), obj);
```

Comment:
We have to remove prototype bean from rootContextMap after postProcessors. Or we can use another map for storing temporary beans, and fully initialised objects would be stored into context after postProcessors.

Reply:
added new instance creation for prototype case